### PR TITLE
Add time constraint between signature and verification

### PR DESCRIPTION
**Pull request by eraffel-MDSol**

### CODE REVIEW

File: src/Medidata.MAuth.Core/MAuthAuthenticator.cs [hunk 1]
```diff
@@ -109,10 +114,22 @@ private async Task<bool> Authenticate(HttpRequestMessage request, MAuthVersion v
                     authInfo.ApplicationUuid.ToString(),
                     () => SendApplicationInfoRequest(authInfo.ApplicationUuid)).ConfigureAwait(false);
 
+            if (!IsSignatureTimeValid(authInfo.SignedTime))
+            {
+                return false;
+            }
```
Comment (johnduhart) [inline src/Medidata.MAuth.Core/MAuthAuthenticator.cs:120]: In my Go implementation I put this check before the public key lookup given it's a cheap check.

**Comment by johnduhart** [comment]
@mdsol/enablement-dotnet

File: src/Medidata.MAuth.Core/MAuthAuthenticator.cs (hunk 1 above)
Comment (eraffel-MDSol) [reply src/Medidata.MAuth.Core/MAuthAuthenticator.cs:116]: https://github.com/mdsol/mauth-client-dotnet/pull/96/commits/9ac0976695bf4802e41d9f271211381e259a5456

File: src/Medidata.MAuth.Core/MAuthAuthenticator.cs [hunk 2]
```diff
@@ -104,15 +109,27 @@ private async Task<bool> Authenticate(HttpRequestMessage request, MAuthVersion v
 
             var mAuthCore = MAuthCoreFactory.Instantiate(version);
             var authInfo = GetAuthenticationInfo(request, mAuthCore);
-            
+
+            if (!IsSignatureTimeValid(authInfo.SignedTime))
+            {
+                return false;
```
Comment (prajon84) [inline src/Medidata.MAuth.Core/MAuthAuthenticator.cs:115]: Do we need to throw `TimeOutException` and log error similar to other cases ?

Comment (awiesendangermdsol) [reply src/Medidata.MAuth.Core/MAuthAuthenticator.cs:115]: How about a log though? If I had to debug why? Not sure how logging is throughout this library.

Comment (prajon84) [reply src/Medidata.MAuth.Core/MAuthAuthenticator.cs:115]: Yeah at least a logger would be beneficial too.

Comment (Jman420) [reply src/Medidata.MAuth.Core/MAuthAuthenticator.cs:115]: Agreed on logging.

**Review by Jman420 — CHANGES_REQUESTED** [review]

File: src/Medidata.MAuth.Core/MAuthAuthenticator.cs (hunk 2 above)
Comment (eraffel-MDSol) [reply src/Medidata.MAuth.Core/MAuthAuthenticator.cs:115]: https://github.com/mdsol/mauth-client-dotnet/pull/96/commits/bf7b84829178af0a6d14df0fe5ed18f5518c751d

**Review by Jman420 — APPROVED** [review]

**Pull request merged**